File: app/src/main/java/com/softwareoverflow/hiitultimate/database/dao/WorkoutSetDao.java
package com.softwareoverflow.hiitultimate.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Transaction;

import com.softwareoverflow.hiitultimate.database.dto.WorkoutSet;
import com.softwareoverflow.hiitultimate.database.entity.ExerciseTypeEntity;
import com.softwareoverflow.hiitultimate.database.entity.WorkoutSetEntity;

import java.util.List;

@Dao
public abstract class WorkoutSetDao implements BaseDao<WorkoutSetEntity> {

    @Query("Select * FROM workoutSet join exerciseType on exerciseTypeId = fkExerciseTypeId WHERE fkWorkoutId = :workoutId")
    public abstract LiveData<List<WorkoutSet>> loadSetsForWorkout(long workoutId);

    @Transaction
    @Insert
    public abstract void insert(WorkoutSetEntity setEntity, ExerciseTypeEntity typeEntity);
}

File: app/src/main/java/com/softwareoverflow/hiit_trainer/data/dao/WorkoutSetDao.kt
package com.softwareoverflow.hiit_trainer.data.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query
import com.softwareoverflow.hiit_trainer.data.WorkoutSet
import com.softwareoverflow.hiit_trainer.data.entity.WorkoutSetEntity

@Dao
interface WorkoutSetDao :
    BaseDao<WorkoutSetEntity> {

    @Query("SELECT * FROM WorkoutSet where id = :workoutSetId")
    fun getWorkoutSetById(workoutSetId: Long): LiveData<WorkoutSet>
}